### PR TITLE
Add check-version-bump workflow

### LINKED ISSUE #40: Consider adding automation?
GitHub Actions can do a lot of magic stuff for you without much work! Here's some things that a lot of npm packages have that might be interesting to add to this project:
 
1. A test workflow that runs `npm test` on each Node LTS version
2. A publish workflow to auto run `npm publish` with an npm token on each release

### PR DESCRIPTION
**Pull request by jcbhmr**
fixes #40
This PR would...
- [x] Fail when a PR doesn't bump the version

### CODE REVIEW

**Comment by mesqueeb** [comment]
@jcbhmr I wonder if we should still merge this.

I came to the realisation: adding `"publish": false,` to the `"np"` configuration in package.json actually might be enough in our case.

- You can make a release if you want by manually creating a tag & release on Github. Manually writing the release notes.
- I can make a release if I want just as always: `npm run release` which runs `np` to do some minor safety checks (no changed files, correct branch, run tests) then it still creates the Github Release for me automatically in the browser with the automatically added release notes like i'm used to.

The fact that it won't publish from my local machine means the github action now succeeds.
Also made sure tests are ran again before the npm publish, just in case.

So I _think_ that covers my automation wishes? Not sure if we need to force PR authors to bump their own package in this case. : )

What do you think?

**Comment by jcbhmr** [comment]
I still think **warning** (it doesn't block the merge) with a ❌ since you didn't bump the version is a Good Idea™. This is like a very basic version of "semantic release" workflow that doesn't incur any overhead like requiring labels on PRs like "major" and doesn't require "feat(thing): did thing"-style commit conventions.

TLDR: I think that this is still a good thing to merge because it stops silly "forgor version bump 😜" commits AND forces PR authors to consciously think "what kind of change is this?"

File: .github/workflows/check-version-bump.yml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: Check version bump
+on:
+  pull_request:
+    branches: 'production'
+    paths:
+      - src/**
+      - test/**
```
Comment (jcbhmr) [inline .github/workflows/check-version-bump.yml:7]: Should this be here? 🤔 Do test changes require version bumps? Maybe not...

**Comment by mesqueeb** [comment]
I don't think we need this.
`npm run release` will bump the version, add tags, push to github, create a release. Then a github action uploads to npm. So I think all bases are covered and the actual release is in github actions. Thanks for the thoughts though!
